sidechain/consensus/slots/src/lib.rs

```diff
@@ -219,23 +240,18 @@ pub trait SimpleSlotWorker<B: ParentchainBlock> {
 			);
 		}
 
-		let _claim =
-			self.claim_slot(&slot_info.last_imported_parentchain_head, slot, &epoch_data)?;
-
-		// Import and retrieve the parentchain header on which the new sidechain block will be based on.
-		let latest_imported_parentchain_header =
-			match self.import_latest_parentchain_block(&slot_info.last_imported_parentchain_head) {
-				Ok(p) => p,
-				Err(e) => {
-					warn!(
-						target: logging_target,
-						"Failed to import and retrieve latest parentchain block header: {:?}", e
-					);
-					return None
-				},
-			};
-
-		let proposer = match self.proposer(latest_imported_parentchain_header.clone(), shard) {
+		let _claim = self.claim_slot(&latest_parentchain_header, slot, &epoch_data)?;
```

Comment:
So it was important that here at `claim_slot`, we have an updated parentchain header (which was not the case before) ?

Reply:
Yes, exactly. Check out the detailed info: https://github.com/integritee-network/worker/issues/750#issuecomment-1144655779